### PR TITLE
Pr/7748

### PR DESCRIPTION
**Pull request by Srokap**
That finises up https://github.com/Elgg/Elgg/pull/7748, something for @mrclay to squash perhaps.

### CODE REVIEW

File: docs/guides/routing.rst [hunk 1]
```diff
@@ -1,4 +1,90 @@
...
+The ``route`` plugin hook is triggered earlier, before page handlers are called. The
+first URL segment is type of the hook (called identifier). This hook can be used to either modify
+identifier and segments or to take over page rendering completely.
+
+The following code intercepts requests to the page handler for ``customblog`` and internally redirects them
+to the ``blog`` page handler.
+
+.. code:: php
+
+    elgg_register_plugin_hook_handler('route', 'customblog', function($hook, $type, $returnvalue, $params) {
```
Comment (mrclay) [inline docs/guides/routing.rst]: Anon functions (for now) can't be unregistered. Motivation for #7750. Should we put them in docs?

Comment (mrclay) [inline docs/guides/routing.rst]: Otherwise I think this is fine.

Comment (Srokap) [inline docs/guides/routing.rst]: Part of me wants it to stay due to elegance of a solution, though until we figure out good way of unregistering them, we probably should  change it to old style for now.

Comment (Srokap) [inline docs/guides/routing.rst]: Fixed.

**Comment by mrclay** [comment]
Please take a look at #7749. I think it's a bit clearer.

**Pull request closed without merging**